### PR TITLE
snyk: add support for new api

### LINKED ISSUE #9083: [Snyk] Migrating from deprecated V1 API
Our Snyk integration collects vulnerability and audit logs from Snyk and relies on their V1 API, which has now been deprecated. Snyk are encouraging all partners to move to their [V3 API](https://docs.snyk.io/snyk-api#snyk-rest-api) to ensure no interruption to ingestion once the V1 API is fully end of life. 

We need to update our integration to point to the V3 API and ensure all authentication, data collection, normalisation, etc continues to work once on the V3 API. Snyk have agreed to give us an enterprise license in order to carry out the migration.

### PR DESCRIPTION
**Pull request by efd6**
## Proposed commit message

See title.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [ ] 

## Related issues

- Closes #9083
- For #5360

## Screenshots

The legacy data streams' UIs remain the same except for the addition of "(Legacy)" after the title.

Screenshot from 2024-05-22 11-42-06
Screenshot from 2024-05-22 11-43-16
Screenshot from 2024-05-22 11-44-16
Screenshot from 2024-05-22 11-44-46
Screenshot from 2024-05-22 11-45-30

### CODE REVIEW

**Comment by efd6** [comment]
@kcreddy I'd particularly like your input into the issues data stream. I has very similar semantics to ti integrations in that it can have updated content since the backing CVEs and so on can receive updates. The API allows this to be queried, but I have not included this here.

File: packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+                "audit_logs": {
+                    "content": {
+                        "after": {
+                            "name": "elastic-integration"
+                        },
+                        "before": {
+                            "name": "admin.user"
+                        }
+                    },
+                    "org_id": "86054e75-398a-4a1a-9dd8-72d026d8c237",
```
Comment (kcreddy) [inline packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json:27]: Could be used in [`organization.id`]([organization.id](https://www.elastic.co/guide/en/ecs/current/ecs-organization.html#field-organization-id))

File: packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json [hunk 2]
```diff
@@ -0,0 +1,717 @@
+{
+    "expected": [
+        {
+            "@timestamp": "2024-04-15T19:47:21.565Z",
+            "ecs": {
+                "version": "8.12.0"
+            },
+            "event": {
+                "action": "org.edit",
```
Comment (kcreddy) [inline packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json:9]: We could probably have [`event.type: change`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-change) when there's something being edited/modified?

File: packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+            }
+        },
+        {
+            "@timestamp": "2024-04-16T09:46:29.448Z",
+            "ecs": {
+                "version": "8.12.0"
+            },
+            "event": {
+                "action": "org.user.invite_link.create",
+                "original": "{\"content\":{\"openInvitePublicId\":\"5cc83241-4d28-401e-80e7-3a421cee2c03\",\"url\":\"https://app.snyk.io/invite/link/accept?invite=5cc83241-4d28-401e-80e7-3a421cee2c03\\u0026utm_source=link_invite\\u0026utm_medium=referral\\u0026utm_campaign=product-link-invite\\u0026from=link_invite\"},\"created\":\"2024-04-16T09:46:29.448Z\",\"event\":\"org.user.invite_link.create\",\"org_id\":\"86054e75-398a-4a1a-9dd8-72d026d8c237\",\"user_id\":\"b4b324c4-a55c-4cd6-82b8-f96e3b3b8d85\"}"
```
Comment (kcreddy) [inline packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json:77]: Could possibly use [`uri_parts`](https://www.elastic.co/guide/en/elasticsearch/reference/current/uri-parts-processor.html) on `url`?

File: packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+                "id": "b4b324c4-a55c-4cd6-82b8-f96e3b3b8d85"
+            }
+        },
+        {
+            "@timestamp": "2024-04-16T21:54:33.257Z",
+            "ecs": {
+                "version": "8.12.0"
+            },
+            "event": {
+                "action": "org.user.add",
```
Comment (kcreddy) [inline packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json:107]: Could possibly have [`event.type: creation`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-type.html#ecs-event-type-creation)

File: packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+                "id": "14b442f2-02f8-47d4-ba51-1749d3771e2c"
+            }
+        },
+        {
+            "@timestamp": "2024-04-17T01:24:49.748Z",
+            "ecs": {
+                "version": "8.12.0"
+            },
+            "event": {
+                "action": "org.integration.settings.edit",
```
Comment (kcreddy) [inline packages/snyk/data_stream/audit_logs/_dev/test/pipeline/test-snyk-audit.json-expected.json:202]: Probably [`event.category: configuration`](https://www.elastic.co/guide/en/ecs/current/ecs-allowed-values-event-category.html#ecs-event-category-configuration) better describes these events when `action` contains `settings`

File: packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+                        ],
+                        "risk": {
+                            "score": {
+                                "model": "v1",
+                                "value": 150
+                            }
+                        },
+                        "status": "open",
+                        "title": "Integer Overflow or Wraparound",
+                        "type": "package_vulnerability",
```
Comment (kcreddy) [inline packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json:379]: Could this be possibly used on [ `vulnerability.category`](https://www.elastic.co/guide/en/ecs/current/ecs-vulnerability.html#field-vulnerability-category)

File: packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+                        "updated_at": "2024-04-17T01:26:52.003Z"
+                    },
+                    "id": "173bddf2-211c-457a-8a91-52e2e9e3baa7",
+                    "relationships": {
+                        "organization": {
+                            "data": {
+                                "id": "0de7b2d6-c1da-46aa-887e-1886f96770d4",
```
Comment (kcreddy) [inline packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json:386]: Could be copied into `organization.id`

File: packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json [hunk 8]
```diff
@@ -0,0 +1,427 @@
...
+                                        "dependency": {
+                                            "package_name": "expat",
+                                            "package_version": "2.1.0-10.el7_3"
+                                        }
+                                    }
+                                ]
+                            }
+                        ],
+                        "created_at": "2024-04-17T01:26:56.55Z",
+                        "effective_severity_level": "medium",
```
Comment (kcreddy) [inline packages/snyk/data_stream/issues/_dev/test/pipeline/test-snyk-issues.json-expected.json:46]: Could be copied into [`vulnerability.severity`](https://www.elastic.co/guide/en/ecs/current/ecs-vulnerability.html#field-vulnerability-severity)

**Comment by efd6** [comment]
> I think it makes sense to include the latest transform to collect the latest status of CVEs and add dashboards on top of these latest documents. A similar use case I found is with: https://github.com/elastic/integrations/pull/9246#issuecomment-2020234674, and also with github.issues which are also bound to update.

Let's do that in a follow up.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**